wordpress-mobile/WordPress-Android
# Adds: blaze overlay with and without post content

Part of #17912 

## What? 
This PR adds an overlay for the Blaze. This Overlay will be displayed as an introduction page when a user tries the blaze feature. He will be redirected to this page, when he tries the Blaze feature through Dashboard or through Published Posts. Created this PR to act as a skeleton to unblock any other development. There are still some changes which need to be made. Shared in **Known issues and Improvements - Todo** 

## Why?
This overlay acts an introduction to the Blaze feature. This page will be shown when the user tries to promote a post with blaze or when he tries to use the blaze feature from Dashboard or Menu. 

 ## How? 
- [Adds: Activity, Fragment and Viewmodel for blaze view](https://github.com/wordpress-mobile/WordPress-Android/pull/17941/commits/d88b5a619a336a49c84c43654f10a3b7bdf2650a)
- [Adds: the ui states and flow source](https://github.com/wordpress-mobile/WordPress-Android/pull/17941/commits/68da981a3c93a8de661ac7a19faf31c779638121)
- [Adds: Image button component to be used in Blaze overlay](https://github.com/wordpress-mobile/WordPress-Android/pull/17941/commits/60ad608144de7ff4a026f8e3b822651d90fd810a)
- [Replaces: Blaze flow source in utils](https://github.com/wordpress-mobile/WordPress-Android/pull/17941/commits/454016161a8b0a2266aa0c36d77b494884447174)

## Testing?

### Pre-requisistes 
- Go to jetpack app 
- Login with a wp account with Administor privileges 
- Go to **app settings** -> **Debug settings** 
- enable feature flag -> `blaze`

### Test 1 - Published Post -> Blaze overlay 
- Go to app -> Posts 
- Go to Published Posts -> Click on More menu
- Click on Promote with Blaze 
- Verify that the Overlay is shown 
- Verify that post title and URL is shown 
- Note: Post featured content preview will be taken up later 

### Test 2 - Dashboard card -> Blaze overlay 
- Go to app -> Home 
- Click on promote with blaze card 
- Verify that the Overlay is shown 
- Verify that no post details are shown 

### ⚠️  The back navigation is not implemented, Default navigation on Android like back button on the bottom or swipe to back navigation will work 

 ## Screenshots (optional)
| Overlay Light Mode  | Overlay Night Mode | 
|---|---|
|  ![screenshot_blaze_overlay_light_](https://user-images.githubusercontent.com/17463767/220633326-f59926fb-68ef-4e0b-bbf7-a40f9d3cb232.png) |  ![screenshot_blaze_overlay_night_](https://user-images.githubusercontent.com/17463767/220633357-d74991b5-a92c-4221-b7b1-9e8f9a2193fc.png)|  


## Known issues and Improvements - Todo 
- Need to fix the Darkmode and Light mode color changes 
- Need to fix any padding issues 
- Need to implement the logic to show featured image in post thumbnail 
- Implement the X-close button or back navigation 
- Implement the Unit tests for ViewModel and UI states

## Regression Notes
1. Potential unintended areas of impact
Blaze overlay doesn't work as intended 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing 

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
